spartan/releases/rough-rhino/index.ts

```diff
@@ -0,0 +1,419 @@
+    const publicIP = await getPublicIP();
+    spinner.succeed(`Public IP: ${publicIP}`);
+
+    // Load existing config
+    spinner.stopAndPersist({ text: "Loading configuration..." });
+    const currentConfig = existsSync(".env")
+      ? Object.fromEntries(
+          readFileSync(".env", "utf8")
+            .split("\n")
+            .filter(Boolean)
+            .map((line) => line.split("="))
+        )
+      : {};
+
+    if (!options.name) {
+      options.name = await input({
+        message: "Validator Name:",
+        default: currentConfig.name || defaultConfig.name,
+      });
+    }
+
+    if (!options.p2pPort) {
+      options.p2pPort = await input({
+        message: "P2P Port:",
+        default: currentConfig.p2pPort || defaultConfig.p2pPort,
+      });
+    }
+
+    if (!options.port) {
+      options.port = await input({
+        message: "Node Port:",
+        default: currentConfig.port || defaultConfig.port,
+      });
+    }
+
+    if (!options.key) {
+      options.key = await input({
+        message: "Validator Private Key:",
+        required: true,
+      });
+    }
+
+    if (!options.ip) {
+      options.ip = await input({
+        message: "Public IP:",
+        default: publicIP || defaultConfig.ip,
+      });
+    }
+
+    // Restart spinner for saving config
+    spinner.start("Saving configuration...");
+
+    const envContent = Object.entries(options)
+      .map(([key, value]) => `${key}=${value}`)
+      .join("\n");
+
+    writeFileSync(".env", envContent, {
+      encoding: "utf8",
+      flag: "w",
+    });
+
+    spinner.succeed("Environment configured successfully");
+
+    // Generate docker-compose.yml
+    spinner.start("Generating docker-compose configuration...");
+    const composeConfig = `
+name: ${options.name}
+services:
+    validator:
+        network_mode: host
+        restart: unless-stopped
+        env_file:
+            - .env
+        environment:
+            - P2P_UDP_ANNOUNCE_ADDR=${options.ip}:${options.p2pPort}
+            - P2P_TCP_ANNOUNCE_ADDR=${options.ip}:${options.p2pPort}
+            - COINBASE=0xbaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+            - VALIDATOR_DISABLED=false
+            - VALIDATOR_PRIVATE_KEY=${options.key}
+            - SEQ_PUBLISHER_PRIVATE_KEY=${options.key}
+            - L1_PRIVATE_KEY=${options.key}
+            - DEBUG=aztec:*,-aztec:avm_simulator*,-aztec:circuits:artifact_hash,-aztec:libp2p_service,-json-rpc*,-aztec:world-state:database,-aztec:l2_block_stream*
+            - LOG_LEVEL=debug
+            - AZTEC_PORT=${options.port}
+            - P2P_ENABLED=true
+            - L1_CHAIN_ID=1337
+            - PROVER_REAL_PROOFS=true
+            - PXE_PROVER_ENABLED=true
+            - ETHEREUM_SLOT_DURATION=12sec
+            - AZTEC_SLOT_DURATION=36
+            - AZTEC_EPOCH_DURATION=32
+            - AZTEC_EPOCH_PROOF_CLAIM_WINDOW_IN_L2_SLOTS=13
+            - ETHEREUM_HOST=http://35.221.3.35:8545
```

Comment:
Need to update this and the ENR based on https://github.com/AztecProtocol/aztec-packages/pull/10352/files

Reply:
We should also add `PEER_ID_PRIVATE_KEY` as we did in that PR.